api/tasks.go

```diff
@@ -857,6 +857,7 @@ type Vault struct {
 	Env          *bool    `hcl:"env,optional"`
 	ChangeMode   *string  `mapstructure:"change_mode" hcl:"change_mode,optional"`
 	ChangeSignal *string  `mapstructure:"change_signal" hcl:"change_signal,optional"`
+	File         *bool    `mapstructure:"file" hcl:"file,optional"`
```

Comment:
For my application this would work (I don't want the token either on disk nor in the environment). I could imagine that there are users though that need the file on disk, but want to avoid it in the environment (e.g., to exclude it from logging), those people would be in trouble if we change behavior. I have no metrics, if that is a thing, so this is theoretic.

Reply:
Good news, another Nomad user came to us asking for this same sort of thing and they definitely want to have `env` without `file` for their use case. So that's at least one vote for keeping them separate and none against 😁.